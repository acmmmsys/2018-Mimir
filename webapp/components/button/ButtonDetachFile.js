import React from 'react';
import { connect } from 'react-redux';
import { compose, withProps, withHandlers } from 'recompose';
import { detachFile } from 'actions';

import { isEmpty } from 'utils';

import ButtonAction from 'components/button/ButtonAction'

const enhance = compose(
    connect(
        null,
        dispatch => ({
            detachImage(imageId) {
                dispatch(detachFile(imageId))
            },
        }),
    ),
    withProps(props => ({
        label: 'Detach Image',
    })),
    withHandlers({
        onClick: props => event => {
            if (props.imageId) {
                props.detachImage(props.imageId);
            }
        }
    })
);

export default enhance(ButtonAction);