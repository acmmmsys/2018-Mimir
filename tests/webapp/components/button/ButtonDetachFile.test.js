import React from 'react';
import ButtonDetachFile from '../../../../webapp/components/button/ButtonDetachFile'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('ButtonDetachFile does render', () => {
  const wrapper = shallow(
      <ButtonDetachFile store={mockStore()}>This is a test!</ButtonDetachFile>
  );
  expect(wrapper).toMatchSnapshot();
});
