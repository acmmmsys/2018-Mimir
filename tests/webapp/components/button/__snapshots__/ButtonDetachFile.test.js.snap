// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDetachFile does render 1`] = `
<withProps(withHandlers(ButtonAction))
  detachImage={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  This is a test!
</withProps(withHandlers(ButtonAction))>
`;
